artemis/reporting/export.py

```diff
@@ -0,0 +1,273 @@
+
+environment = Environment(loader=BaseLoader(), undefined=StrictUndefined, trim_blocks=True, lstrip_blocks=True)
+OUTPUT_LOCATION = "./output/autoreporter/"
+
+
+def _get_domain_or_ip_only(data: str) -> str:
+    """Extracts domain/ip from URL or host:port string."""
+    if "://" in data:
+        host = urllib.parse.urlparse(data).hostname
+        assert host
+        return host
+    if ":" in data:
+        return data.split(":")[0]
+    return data
+
+
+def _deduplicate_ip_vs_domain_versions(
+    already_seen_reports: List[Report], reports_to_send: List[Report]
+) -> List[Report]:
+    """Skips reports if:
+
+    - a vulnerability is on a URL with an IP and an identical report for a domain with this IP has
+      already been sent,
+    - a vulnerability is on a URL with an IP and an identical report for a domain with this IP is
+      planned to be sent,
+    - a vulnerability is on a URL with a domain and an identical report for a URL with an IP of this domain has
+      already been sent.
+    """
+    already_seen_reports_ip_normal_forms: Set[Any] = set()
+    already_seen_reports_normal_forms: Set[Any] = set()
+    for already_seen_report in already_seen_reports:
+        already_seen_reports_normal_forms.add(already_seen_report.get_normal_form())
+        if not already_seen_report.target_is_ip_address():
+            alternative_with_ip_address = already_seen_report.alternative_with_ip_address()
+            if alternative_with_ip_address:
+                already_seen_reports_ip_normal_forms.add(alternative_with_ip_address.get_normal_form())
+
+    reports_to_send_ip_normal_forms: Set[Any] = set()
+    for report in reports_to_send:
+        if not report.target_is_ip_address():
+            alternative_with_ip_address = report.alternative_with_ip_address()
+            if alternative_with_ip_address:
+                reports_to_send_ip_normal_forms.add(alternative_with_ip_address.get_normal_form())
+
+    reports_to_send_filtered: List[Report] = []
+    for report in reports_to_send:
+        # This is an IP report and is an ip-converted version of an existing report
+        if report.target_is_ip_address() and report.get_normal_form() in already_seen_reports_ip_normal_forms:
+            continue
+        # This is not an IP report but an IP report for the same has already been sent
+        if not report.target_is_ip_address():
+            alternative_with_ip_address = report.alternative_with_ip_address()
+            if (
+                alternative_with_ip_address
+                and alternative_with_ip_address.get_normal_form() in already_seen_reports_normal_forms
+            ):
+                continue
+        # This is an IP report but we have a non-ip version to send
+        if report.target_is_ip_address() and report.get_normal_form() in reports_to_send_ip_normal_forms:
+            continue
+        reports_to_send_filtered.append(report)
+    return reports_to_send_filtered
```

Comment:
No need to type name in variable names. This code is very unreadable. Perhaps split into smaller functions?